src/installer/corehost/cli/bundle/runner.h

```diff
@@ -37,8 +37,22 @@ namespace bundle
 
         static const runner_t* app() { return (const runner_t*)the_app; }
 
+        // Obtain the bundle_probe function (encoded as a string) to be passed to the runtime.
+        static pal::string_t get_bundle_probe();
+
     private:
 
+        // Probe the app-bundle for the file 'path' and return its location ('offset', 'size') if found.
+        // This method is intnded to be used by the runtime to probe for bundled assemblies
+        // This method assumes that the currently executing app is a single-file bundle.
+        //
+        // bundle_probe recieves its path argument as wchar_t* instead of pal::char_t*, because:
+        // * The host uses Unicode strings on Windows and UTF8 strings on Unix
+        // * The runtime uses Unicode strings on all platforms
+        // Using a unicode encoded path presents a uniform interface to the runtime
+        // and minimizes the number if Unicode <-> UTF8 conversions necessary.
+        static bool __stdcall bundle_probe(const wchar_t* path, int64_t* offset, int64_t* size);
```

Comment:
I would make this a top-level/global function and not a member function - to make it clear it's effectively an exported API. Or maybe even better - do the same which hostpolicy already does for functions like `get_delegate` and similar. Those are also effectively exported APIs which are returned as function pointers. Currently they are written as hidden inside an anonymous namespace and don't need header/cpp duplicate declarations.

Also currently the header and cpp have slightly different declarations of the function - which feels wrong.